pkg/trace/config/apply.go

```diff
@@ -201,6 +202,11 @@ func (c *AgentConfig) applyDatadogConfig() error {
 			}
 		}
 	}
+	prevEnv := c.DefaultEnv
+	c.DefaultEnv = traceutil.NormalizeTag(c.DefaultEnv)
+	if c.DefaultEnv != prevEnv {
+		log.Debugf("Normalized DefaultEnv from %q to %q", prevEnv, c.DefaultEnv)
```

Comment:
Actually, the env may come from other places too, not just my suggestion, so maybe it's fine the way you have it. Or maybe saying "default env" instead of "DefaultEnv".

Reply:
I thought it'd be useful because the original log is emitted pre-normalization. You're right that we don't report on env normalization from traces. But in this specific case we already have debug logs, so I felt like it'd be right to have a "follow-up" log for clarity.